### PR TITLE
zsh-navigation-tools 2.0.5 (new formula)

### PR DESCRIPTION
**Pull request by psprint**
This set of tools provides advanced history searcher for Zsh. It uses Curses, supports color themes, etc. but most of all: is multi-word. There is "zsh-history-substring-search" package and this is similar. Full description:

Set of tools like n-history – multi-word history searcher, n-cd – directory bookmark manager, n-kill – htop like kill utility, and more. Feature highlights include incremental multi-word searching, ANSI coloring, unique mode, horizontal scroll, non-selectable elements, grepping and various integrations with Zsh.
- n-history: the full-screen history searcher
- n-kill - browses processes list, allows quick searching and sending signal to selected processes
- n-panelize - loads output of given command into the list for browsing and multi-word searching
- n-cd - browses dirstack and bookmarked directories, allows to enter selected directory
- n-aliases - browses aliases, relegates editing to vared
- n-functions - browses functions, relegates editing to zed or vared
- n-env - browses environment, relegates editing to vared
- n-options - browses options, allows to toggle their state

This is written in pure Zsh, homepage is: https://github.com/psprint/zsh-navigation-tools

### CODE REVIEW

**Comment by psprint** [comment]
Should I fix the error:

```
Error Message
failed: brew audit zsh-navigation-tools --strict --online

Stacktrace
        Error: 1 problem in 1 formula
zsh-navigation-tools:
  * Stable: version 2.0.5 is redundant with version scanned from URL
```

I saw other packages do use "version" field, should I remove it?

File: Formula/zsh-navigation-tools.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    You will also need to force reload of your .zshrc:
+
+      source ~/.zshrc
+
+    You can see all the tools provided by entering "n-" and hitting <Tab> to complete.
+    N-history, the incremental multi-word history searcher, is bound to Ctrl-R.
+    Configuration files are in ~/.config/znt
+    EOS
+  end
```
Comment (UniqMartin) [inline Formula/zsh-navigation-tools.rb]: Please remove the caveats. These instructions are not Homebrew-specific and thus don't belong in here.

File: Formula/zsh-navigation-tools.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  test do
+    # This compiles package's main file
+    # Zcompile is very capable of detecting syntax errors
+    cp "#{pkgshare}/n-list", "."
```
Comment (UniqMartin) [inline Formula/zsh-navigation-tools.rb]: Make this `cp pkgshare/"n-list", testpath` instead.

**Comment by UniqMartin** [comment]
> I saw other packages do use "version" field, should I remove it?

The `version` line is only needed if Homebrew fails to detect the version from the download URL.

> Update: I have fixed the error (by removing version field)

:+1: That's the right fix.

File: Formula/zsh-navigation-tools.rb (hunk 1 above)
Comment (psprint) [inline Formula/zsh-navigation-tools.rb]: I removed description of software. What's left is instruction to run it, like in zsh-syntax-highlighting package. Can it be that way?

File: Formula/zsh-navigation-tools.rb (hunk 2 above)
Comment (psprint) [inline Formula/zsh-navigation-tools.rb]: Done

**Comment by psprint** [comment]
Hello. Not sure if commenting outdated diffs triggers notifications. I've updated the formula, hope everything is in check now

Best regards,
Sebastian Gniazdowski

**Comment by UniqMartin** [comment]
> Not sure if commenting outdated diffs triggers notifications. I've updated the formula, hope everything is in check now

It does. 😸 I made some minor stylistic adjustments, but otherwise it's fine.

Merged in 7e265f9e4038c44439b4cf4f2701a9a4a1b622dc. Thank you for this contribution to Homebrew, @psprint! :tada:

**Comment by psprint** [comment]
Thank you for accepting the formula, feels nice :)

**Pull request closed without merging**